modules/registrar/save.c

```diff
@@ -950,6 +951,11 @@ int _remove(struct sip_msg *msg, void *udomain, str *aor_uri, str *match_ct,
 		goto out_unlock;
 	}
 
+	if (bflags_s) {
+		bflags = get_flag_id_by_name(FLAG_TYPE_BRANCH, bflags_s->s, bflags_s->len);
+		bflags = 1<<bflags;
+	}
```

Comment:
There is some potential for improvement here:

1. this repetitive flag parsing shouldn't be done at runtime (**N** times), rather during the fixup phase (**1** time).  And even if the flags were to arrive via a variable, the function interface will detect this case and correctly invoke the fixup _per function call_.
2. if the original intent was to allow more flags, then `flag_list_to_bitmask()` should have been used instead of `get_flag_id_by_name()`.  Going down the rabbit hole (credits @bogdan-iancu):  should the multiple flags be AND'ed or OR'ed?  As long as the docs clarify this behavior, anything is OK.  I see the current implementation opted for "OR".

Reply:
Addressed via 7fce69b4429f806e3f91186ed86ae2765a517704